### ISSUE TITLE
[remocon] Apps are not shut down

### ISSUE THREAD

**Issue by bit-pirate** [robotics-in-concert/rocon_qt_gui#108]
When shutting down the remocon, I can see that it is trying to shut down all running apps:

```
[debug] InteractiveClient : shutting down all apps
InteractiveClient : updated app list- {}
[remocon_info] publish remocon status
[debug] InteractiveClient : has shutdown.
rocon_remocon: Main end
[info] RemoconMain: Destroy
```

Unfortunately, this has no effect. The app manager does not receive the service call to stop the running rapp.

**Comment by bit-pirate** [comment]
Now, with the "stopping app bug" (https://github.com/robotics-in-concert/rocon_tools/issues/38) fixed, the remocon does actually shutdown the app, if one is running, i.e. either when hitting the "stop interactions" or the "back" button.

So, the only confusing part is, that, if no rapp is running, the remocon still claims to shutdown the rapps. This might be just a cosmetics issue.

Unfortunately, I just found another related issue: If a rapp is running and the remocon is closed by hitting the x button (top right), it shuts down without stopping the rapps. This should be fixed, because otherwise the only way to recover from this situations is to restart the master.

**Comment by stonier** [comment]
First point was just a cosmetics issue. 

Second point is a real issue and still remains.

**Comment by stonier** [comment]
@bit-pirate - Im actually seeing different behaviour.
- I shut down the remocon
- Rapp and Qt Interaction keep running
- I close the Qt Interaction
- Rapp automagically stops

This is expected (at least for me). Do you want to show me what you were getting tomorrow?

**Comment by stonier** [comment]
> f a rapp is running and the remocon is closed by hitting the x button (top right), it shuts down without stopping the rapps.

This is fixed in 2415e6646031636801e6e9e57374b7118eec0e22

**Comment by stonier** [comment]
All the other permutations of things that can go wrong are much changed (and should be fixed) in the role-interaction list switching update on the weekend (#118), so all should be good now.

Reopen if necessary.
